etna/core/mixins.py

```diff
@@ -26,6 +28,27 @@ def __repr__(self):
                 args_str_representation += f"{arg} = {repr(value)}, "
         return f"{self.__class__.__name__}({args_str_representation})"
 
+    def to_dict(self):
+        """Collect all information about etna object in dict."""
+        init_args = inspect.signature(self.__init__).parameters
+        params = {}
+        for arg, _ in init_args.items():
+            value = self.__dict__[arg]
+            if isinstance(value, BaseMixin):
+                params[arg] = value.to_dict()
+            else:
+                if isinstance(value, BaseEstimator):
```

Comment:
It seems we can eliminate nested if here.

It's typical switch case pattern without nesting.